Use more smart pointers in StyleScope.h

cdumez opened the pull request:
#### 207c7c9d8ebd6cff491d016db47b46e7a009b8f5
<pre>
Use more smart pointers in StyleScope.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=261983">https://bugs.webkit.org/show_bug.cgi?id=261983</a>

Reviewed by Darin Adler.

* Source/WebCore/css/CSSStyleSheet.h:
* Source/WebCore/css/StyleSheetContents.cpp:
(WebCore::StyleSheetContents::StyleSheetContents):
* Source/WebCore/css/StyleSheetContents.h:
* Source/WebCore/style/StyleInvalidator.cpp:
(WebCore::Style::shouldDirtyAllStyle):
(WebCore::Style::Invalidator::Invalidator):
(WebCore::Style::m_dirtiesAllStyle):
* Source/WebCore/style/StyleInvalidator.h:
* Source/WebCore/style/StyleScope.cpp:
(WebCore::Style::Scope::analyzeStyleSheetChange):
* Source/WebCore/style/StyleScope.h:

Canonical link: <a href="https://commits.webkit.org/268375@main">https://commits.webkit.org/268375@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/459f8315b6bfcf9ae8cf305beb9695936062f688

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19517 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19936 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20544 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21407 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18253 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19753 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23199 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20078 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19857 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19733 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19756 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16962 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22263 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16943 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17750 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24065 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18004 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17925 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22037 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18533 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15702 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17677 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4664 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22033 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18361 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/StyleSheetContents.h:
```diff
@@ -45,7 +46,7 @@ class StyleRuleNamespace;
 enum class CachePolicy : uint8_t;
 
-class StyleSheetContents final : public RefCounted<StyleSheetContents>, public CanMakeWeakPtr<StyleSheetContents> {
+class StyleSheetContents final : public RefCounted<StyleSheetContents>, /*public CanMakeWeakPtr<StyleSheetContents>,*/ public CanMakeCheckedPtr {
```
Please don’t forget to delete the commented-out part here. Maybe also delete the include of `WeakPtr.h`?

cdumez replied:
Oh wow. Sorry about that. I really miss the pretty-diff that `webkit-patch` was showing before uploading the patch. I would always catch mistakes like these thanks to it.

darinadler commented on Source/WebCore/style/StyleScope.h:
```diff
@@ -139,7 +139,7 @@ class Scope : public CanMakeWeakPtr<Scope> {
 
     struct QueryContainerUpdateContext {
-        HashSet<Element*> invalidatedContainers;
+        HashSet<CheckedPtr<Element>> invalidatedContainers;
```
Can we use `CheckedRef` instead?

darinadler commented on Source/WebCore/style/StyleScope.h:
```diff
@@ -210,7 +210,7 @@ class Scope : public CanMakeWeakPtr<Scope> {
     Timer m_pendingUpdateTimer;
 
-    mutable HashSet<const CSSStyleSheet*> m_weakCopyOfActiveStyleSheetListForFastLookup;
+    mutable HashSet<CheckedPtr<const CSSStyleSheet>> m_weakCopyOfActiveStyleSheetListForFastLookup;
```
Can we use `CheckedRef` instead?

cdumez committed:
Use more smart pointers in StyleScope.h
https://bugs.webkit.org/show_bug.cgi?id=261983

Reviewed by Darin Adler.

* Source/WebCore/css/CSSStyleSheet.h:
* Source/WebCore/css/StyleSheetContents.cpp:
(WebCore::StyleSheetContents::StyleSheetContents):
* Source/WebCore/css/StyleSheetContents.h:
* Source/WebCore/style/StyleInvalidator.cpp:
(WebCore::Style::shouldDirtyAllStyle):
(WebCore::Style::Invalidator::Invalidator):
(WebCore::Style::m_dirtiesAllStyle):
* Source/WebCore/style/StyleInvalidator.h:
* Source/WebCore/style/StyleScope.cpp:
(WebCore::Style::Scope::analyzeStyleSheetChange):
* Source/WebCore/style/StyleScope.h:

Canonical link: https://commits.webkit.org/268375@main
